controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -24,19 +25,45 @@ const (
 // https://github.com/linkerd/linkerd2/issues/2204
 
 type (
-	// Address represents an individual port on an specific pod.
-	Address struct {
+
+	// Address represents an individual port on a specific endpoint.
+	// This endpoint might be associated with a Pod, a Headless service
```

Comment:
There are some existing references to "Headless" services in various comments.  These are incorrect.  What they refer to as "headless services" are actually services which have endpoints that do not correspond to pods.  We should remove these misleading comments and variable names.

A headless service is one without a clusterIP.  Linkerd does not interact with the clusterIP of a service and does not care if a service is headless or not.